### PR TITLE
[VET-3766][VET-4297] Update addCredentials to support Atlan and Purview

### PR DESCRIPTION
**Pull request by hasibhassan**
VET-3766 VET-4297

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -1396,10 +1396,10 @@ declare namespace Stardog {
          * Adds a provider credential
          *
          * @param {Connection} conn the Stardog server connection
-         * @param {User} user the username and password to be stored
+         * @param {Credentials} credentials the credentials to be stored
```
Comment (joshhk72) [inline lib/index.d.ts:1399]: Need to also add `Credentials` type under `catalog` namespace (similar to how `DataSourceQuery` is under `dataSources` namespace.

File: lib/catalog.js [hunk 2]
```diff
@@ -22,13 +22,12 @@ const reload = (conn, provider) => {
...
 
-const addCredential = (conn, user, label) => {
+const addCredential = (conn, credentials, label) => {
   const headers = conn.headers();
   headers.set('Content-Type', 'application/json');
 
   const body = {
-    username: user.username,
-    password: user.password,
+    ...credentials,
```
Comment (joshhk72) [inline lib/catalog.js:30]: Works for me, but should we allow the user to set whatever on this? (I don't know if the way we do it normally is from inertia, or security, or etc.).

**Review by joshhk72 — APPROVED** [review]
LGTM, just some comments + small doc change

**Review by anneeb — APPROVED** [review]

**Comment by anneeb** [comment]
released in https://github.com/stardog-union/stardog.js/releases/tag/v5.3.1

**Pull request merged**